### PR TITLE
bugfix: implement DiffSuppressFunc for chart list of log dashboard and pass required fields on update

### PR DESCRIPTION
**Pull request by cofyc**
**What it does**

- implement DiffSuppressFunc to compare chart list semantically
  - JSON fields are unordered after marshaling, we cann't compare the raw string
- dashboard_name and char_list fields must be passed to the UPDATE API every time even if they are not changed
  - if we omit dashboard_name if it's not changed, it will be updated to dashboard id
  - chart_list is required field (the update request will fail)

### CODE REVIEW

**Comment by cofyc** [comment]
@xiaozhu36 PTAL

File: alicloud/resource_alicloud_log_dashboard.go [hunk 1]
```diff
@@ -38,8 +41,9 @@ func resourceAlicloudLogDashboard() *schema.Resource {
 				Optional: true,
 			},
 			"char_list": {
-				Type:     schema.TypeString,
-				Required: true,
+				Type:             schema.TypeString,
+				DiffSuppressFunc: jsonPolicyDiffSuppress,
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_log_dashboard.go:45]: There is a stable diffsuppressfunc to support it and please refer to: https://github.com/aliyun/terraform-provider-alicloud/blob/master/alicloud/resource_alicloud_ram_policy.go#L67

**Review by xiaozhu36 — CHANGES_REQUESTED** [review]

File: alicloud/resource_alicloud_log_dashboard.go (hunk 1 above)
Comment (cofyc) [reply alicloud/resource_alicloud_log_dashboard.go:45]: `compareJsonTemplateAreEquivalent` can't work here. In my testing, `terrarform apply` show changes every time. 

Before comparing the chart lists, we must sort the charts.

Comment (cofyc) [reply alicloud/resource_alicloud_log_dashboard.go:45]: I guess the alicloud API reordered the chart list, so after the configurations are applied to the alicloud,  the`chart_list` may diff from the user configuration in the next `terraform apply` (using compareJsonTemplateAreEquivalent). 

However, the order of the chart list does not matter. I think it's better to sort the charts before comparing in this case.

**Pull request merged**